### PR TITLE
docs: Fixes broken link on Porter Docs

### LINKED ISSUE #2820: Broken link redirecting to a 404 page
## Describe the bug

While trying to learn about porter and going through its documentation, on clicking **[Understand the difference between imperative commands and desired state](https://getporter.org/end-users/installations/)**, User is getting directed to a page which don't exist

Screenshot from 2023-07-12 13-11-54

Screenshot from 2023-07-12 13-11-29

## Version
[ porter v1.0.14 (0e739d88) ]

### PR DESCRIPTION
**Pull request by itsaviral2609**
# What does this change

Broken link identified on Porter documentation site has been 'fixed Understand the difference between imperative commands and desired state

Screenshot from 2023-07-17 20-02-07

Screenshot from 2023-07-17 20-02-20

# What issue does it fix
Closes #2820

# Notes for the reviewer

Please review my PR :smile: 

# Checklist
- [ ] Did you write tests?
- [x] Did you write documentation?
- [ ] Did you change porter.yaml or a storage document record? Update the corresponding schema file.
- [ ] If this is your first pull request, please add your name to the bottom of our [Contributors][contributors] list. Thank you for making Porter better! 🙇‍♀️

# Reviewer Checklist
* Comment with /azp run test-porter-release if a magefile or build script was modified
* Comment with /azp run porter-integration if it's a non-trivial PR

[contributors]: https://getporter.org/src/CONTRIBUTORS.md

### CODE REVIEW

File: .github/workflows/changelog-flow.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: master
+          fetch-depth: 0
+          token: ${{ secrets.GH_PAT }}
```
Comment (troy0820) [inline .github/workflows/changelog-flow.yml:15]: Not sure if we have this, but we would have to wait for the maintainer to check this.

File: .github/workflows/changelog-flow.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    types: [published]
+
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: master
```
Comment (troy0820) [inline .github/workflows/changelog-flow.yml:13]:
```suggestion
          ref: main
```
Should this be main?

File: .github/workflows/changelog-flow.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      - name: Generate changelog
+        run: conventional-changelog -p angular -i CHANGELOG.md -s -r 0
+
+      - name: Commit and push changes
+        run: |
+          git config --global user.name 'GitHub Actions'
+          git config --global user.email 'actions@github.com'
+          git add CHANGELOG.md
+          git commit -m "Update Changelog" || echo "No changes to commit"
+          git push origin master
```
Comment (troy0820) [inline .github/workflows/changelog-flow.yml:34]:
```suggestion
          git push origin main
```

**Review by troy0820 — CHANGES_REQUESTED** [review]
Some quick changes and a question for the secret that the maintainers can answer.

File: .github/workflows/changelog-flow.yml (hunk 1 above)
Comment (itsaviral2609) [reply .github/workflows/changelog-flow.yml:15]: https://github.com/getporter/porter/blob/main/.github/workflows/close-issues.yaml ig they have used secrets in other workflows too!
Nevertheless, will wait for maintainers to have  a say on this!

**Comment by schristoff** [comment]
Hi - this is an annoying ask and I'm sorry in advance - could you separate out the broken links PR and the changelog PR? This way we can nit and have the ability to revert the changelog changes if needed without having to worry about messing with the changes to the docs.
Thank you so much!

**Comment by itsaviral2609** [comment]
@schristoff  Sure! I will make 2 separate PRs for the above issues!
I will make the desired changes as discussed above!

**Pull request closed without merging**